app_dart/lib/src/request_handlers/refresh_chromebot_status.dart

```diff
@@ -80,27 +83,48 @@ class RefreshChromebotStatus extends ApiRequestHandler<Body> {
     /// [builderNumberList] when luci rerun happens, and update [devicelabTask]
     /// status when the status of latest luci run changes.
     for (FullTask datastoreTask in datastoreTasks) {
-      if (luciTasksMap.containsKey(datastoreTask.commit.sha)) {
-        final List<LuciTask> luciTasks = luciTasksMap[datastoreTask.commit.sha];
-        final String buildNumberList =
-            luciTasks.reversed.map((LuciTask luciTask) => luciTask.buildNumber.toString()).toList().join(',');
-        if (buildNumberList != datastoreTask.task.buildNumberList ||
-            luciTasks.first.status != datastoreTask.task.status) {
-          final Task update = datastoreTask.task;
-          update.status = luciTasks.first.status;
-          update.buildNumberList = buildNumberList;
-          update.builderName = builder.name;
-          update.luciBucket = 'luci.flutter.prod';
-          await datastore.insert(<Task>[update]);
-          // Save luci task record to BigQuery only when task finishes.
-          if (update.status == Task.statusFailed || update.status == Task.statusSucceeded) {
-            await _insertBigquery(update);
-          }
+      final String commitSha = datastoreTask.commit.sha;
+      if (!luciTasksMap.containsKey(commitSha)) {
+        continue;
+      }
+      final List<LuciTask> luciTasks = luciTasksMap[commitSha];
+      final String buildNumberList =
+          luciTasks.reversed.map((LuciTask luciTask) => luciTask.buildNumber.toString()).toList().join(',');
+      final LuciTask latestLuciTask = luciTasks.first;
+      if (buildNumberList != datastoreTask.task.buildNumberList || latestLuciTask.status != datastoreTask.task.status) {
+        final Task update = datastoreTask.task;
+        update.status = latestLuciTask.status;
+
+        if (_shouldRerunTask(latestLuciTask, update.attempts)) {
+          await luciBuildService.rescheduleProdBuild(
+            commitSha: commitSha,
+            builderName: latestLuciTask.builderName,
+          );
+          update.status = Task.statusNew;
+          update.attempts += 1;
+        }
+
+        update.buildNumberList = buildNumberList;
+        update.builderName = builder.name;
+        update.luciBucket = 'luci.flutter.prod';
+        await datastore.insert(<Task>[update]);
+        // Save luci task record to BigQuery only when task finishes.
+        if (update.status == Task.statusFailed || update.status == Task.statusSucceeded) {
+          await _insertBigquery(update);
         }
       }
     }
   }
 
+  /// Auto-rerun Mac builders with `infra failure`.
+  ///
+  /// This is a workaround for -9 retcode issue: https://github.com/flutter/flutter/issues/68322.
+  bool _shouldRerunTask(LuciTask luciTask, int attempts) {
+    return luciTask.builderName.contains('Mac') &&
+        luciTask.status == Task.statusInfraFailure &&
+        attempts < config.maxTaskRetries;
```

Comment:
I'd limit to one retry only. Also how do we know that we are explicitly re-running failures with exit code -9

Reply:
> I'd limit to one retry only. 
Done.
> Also how do we know that we are explicitly re-running failures with exit code -9
Please refer to the PR description. We are rerunning Mac `infra-failure` builds since it covers all `-9` ones, and there is not an easy way to specifically filter out `-9` ones.

Reply:
What is the plan to prevent hiding real flakes that report themselves as infra_failures? additionally  what is the plan to only retry things that we understand and already triaged? as it is this PR will be adding a lot of technical debt, we can land it if this is temporary and we already have a plan on how to make the retries more specific and configurable.

Reply:
Yeah, it may hide real flakes if we rerun all infra failure builds.
To avoid potential technical debt, I updated the rerun logic to make it more specific for our -9 use case:
1) rerun all Mac builders (without shards) failing with exact `-9 retcode` in `summaryMarkdown`
2) rerun Mac builders with shards failing with `summaryMarkdown` equal "Step('display builds.build(s) failed') (retcode: 1)".

Reply:
Awesome, thanks!